modules/lib/zsh.nix

```
@@ -7,9 +7,18 @@ rec {
     if builtins.isBool v then
       if v then "true" else "false"
     else if builtins.isString v then
-      ''"${v}"''
+      lib.escapeShellArg v
     else if builtins.isList v then
-      "(${lib.concatStringsSep " " (map toZshValue v)})"
+      let
+        shell = import ./shell.nix { inherit lib; };
+        content = shell.formatShellArrayContent (map toString v);
+      in
+      if content == "" then
+        "()"
+      else if lib.hasInfix "\n" content then
+        "(${content})"
+      else
+        "(${content})"
```

Unless I missed something, this is all equivalent to `"(${content})"`.

```suggestion
      "(${content})"